lib/util.js

```diff
@@ -649,6 +650,8 @@ exports.inherits = function(ctor, superCtor) {
 exports._extend = function(origin, add) {
   // Don't do anything if add isn't an object
   if (add === null || typeof add !== 'object') return origin;
+  // For compatibility with Url objects
+  if (add instanceof Url) add = add.toJSON();
```

Comment:
The alternatives are just as hacky and probably worse:

1. Making new internal extend that supports inherited properties
2. Adding this instanceof check at all call sites (same logic, just more duplication)